java-manta-client/src/main/java/com/joyent/manta/util/ConcurrentWeakIdentityHashMap.java

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
```

Comment:
Per the license I think we are going to need to preserve the `NOTICE` file of this work.  Maybe make a followup ticket that we do before 3.0 is cut?